src/file.c

```diff
@@ -290,6 +290,26 @@ mrb_file_flock(mrb_state *mrb, mrb_value self)
 }
 #endif
 
+static mrb_value
+mrb_file_s_symlink(mrb_state *mrb, mrb_value klass)
+{
+#if defined(_WIN32) || defined(_WIN64)
+  mrb_raise(mrb, E_NOTIMP_ERROR, "symlink is not supported on this platform");
+#else
+  mrb_value from, to;
+  const char *src, *dst;
+
+  mrb_get_args(mrb, "SS", &from, &to);
+  src = mrb_string_value_cstr(mrb, &from);
```

Comment:
`mrb_string_value_cstr()` can only be used on the string that is not a *shared string*.  For example, `mruby -e 'a="abcde"; p File.symlink(a[0,3], "b")'` fails because `a[0,3]` is a shared string and `mrb_string_value_cstr()` raises an exception.

I've committed the fix on https://github.com/iij/mruby-io/commit/c9a4091b83b06b20d8e9178ce748d5b101925a2e .  Thank you for the PR!